passengerfeedback/passenger_client/main.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	pb "../proto"
+	"context"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"google.golang.org/grpc"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	address = "localhost:50051"
+)
+
+var client pb.FeedbackClient
+
+func handleAddFeedback(c *gin.Context) {
+	var argument struct {
+		PassengerId string
+		BookingCode string
+		Feedback    string
+	}
+	err := c.BindJSON(&argument)
+	if err != nil {
+		c.String(400, "invalid param \n")
+		log.Print(err)
+		return
+	}
+	if len(strings.TrimSpace(argument.PassengerId)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing passengerId"})
+		return
+	}
+	passengerId, err := strconv.Atoi(argument.PassengerId)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "PassengerId must be a number"})
+		return
+	}
+	if len(strings.TrimSpace(argument.BookingCode)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing bookingCode"})
+		return
+	}
+	if len(strings.TrimSpace(argument.Feedback)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing feedback"})
+		return
+	}
+	res, err := addPassengerFeedback(argument.BookingCode, int32(passengerId), argument.Feedback)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Something happens"})
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_SUCCESS, "message": res.Msg, "data": res.Data})
+}
+
+func handleGetFeedbackByPassengerId(c *gin.Context) {
+	strPassenger, b := c.GetQuery("passengerId")
+	if !b || len(strings.TrimSpace(strPassenger)) == 0 {
+		c.JSON(http.StatusOK, gin.H{"errorCode": pb.Error_FAIL, "message": "Missing passengerId"})
+		return
+	}
+	passengerId, err := strconv.Atoi(strPassenger)
```

Comment:
we have too many duplicated params validation logic, better to move them to common package and re-use